xonsh/aliases.py

```diff
@@ -254,7 +254,8 @@ def source_cmd(args, stdin=None):
     args[0] = fpath if fpath else args[0]
     if not os.path.isfile(args[0]):
         raise FileNotFoundError(args[0])
-    prevcmd = 'call "{}"\necho off'.format(args[0])
+    prevcmd = 'call "{}"'.format('" "'.join(args))
```

Comment:
Yes I also think Popen handles the argv part. But it can't rely on that here since everything is written to a batch file. I will make a fix.